homeassistant/helpers/singleton.py

```diff
@@ -62,6 +77,45 @@ async def async_wrapped(hass: HomeAssistant) -> Any:
 
             return cast(_T, obj_or_evt)
 
-        return async_wrapped  # type: ignore[return-value]
+        return async_wrapped
 
     return wrapper
+
+
+async def _test_singleton_typing(hass: HomeAssistant) -> None:
```

Comment:
I suppose this should go into a test file in tests instead

Reply:
The test files aren't type checked. At some point, it might make sense to have a separate test suite for type tests as there are one or two more files these tests but I don't think it's necessary just yet.